src/backends/tensorrt/plan_backend.cc

```diff
@@ -3044,6 +3041,21 @@ PlanBackend::Context::Run(
     nvinfer1::Dims dims;
     dims = citr->second.context_->getBindingDimensions(binding_index);
 
+    // Wait for model execution to process the output tensors with pinned memory
+    // address if zero-copy is supported. Otherwise use device memory address,
+    // so that the memory copies perform asynchronously and wait for model
+    // execution.
+    void* buffer = nullptr;
+    if (zero_copy_support_) {
+      buffer = io_binding_info.buffer_;
+      io_binding_info.memory_type_ = TRITONSERVER_MEMORY_CPU_PINNED;
```

Comment:
You shouldn't overwrite anything in `io_binding_info` during run, this information should be able to set once for all during initialization. You may need to change the information that `io_binding_info` carries if needed, but I think what it holds for this change is the device pointer used for TRT enqueue and the buffer / mem type / id (pinned in this case) for process tensor.

Reply:
Yes I had already set this earlier. These updates are not needed.